erpnext/hr/doctype/salary_slip/salary_slip.py

```diff
@@ -815,6 +819,23 @@ def update_status(self, salary_slip=None):
 				timesheet.set_status()
 				timesheet.save()
 
+	def update_loans(self, status):
+		for loan in self.get_loan_details():
+			doc = frappe.get_doc("Loan", loan.name)
+			if status == "Submit":
+				frappe.db.set_value("Repayment Schedule", loan.repayment_name, "paid", 1)
+				doc.reload()
+				doc.total_amount_paid += loan.total_payment
+				total_pay = doc.total_payment - loan.total_payment
+			elif status == "Cancel":
+				frappe.db.set_value("Repayment Schedule", loan.repayment_name, "paid", 0)
+				doc.reload()
+				doc.total_amount_paid -= loan.total_payment
+				total_pay = doc.total_payment + loan.total_payment
+
+			doc.submit()
```

Comment:
`submit()` is misleading as loan is already submitted. What's the need of this?

Reply:
done